diff -urN quake3.orig/code/sys/sys_main.c quake3/code/sys/sys_main.c
--- quake3.orig/code/sys/sys_main.c	2019-07-29 18:40:49.424888973 +0200
+++ quake3/code/sys/sys_main.c	2019-07-29 18:51:00.376127397 +0200
@@ -500,7 +500,7 @@
 Sys_LoadDll
 
 First try to load library name from system library path,
-from executable path, then fs_basepath.
+from executable path, QUAKELIBDIR, then fs_basepath.
 =================
 */
 
@@ -542,6 +542,25 @@
 			Com_Printf("Skipping trying to load \"%s\" from \"%s\", file name is too long.\n", name, topDir);
 		}
 
+#if defined(QUAKELIBDIR)
+		if(!dllhandle)
+		{
+			char libPath[MAX_OSPATH];
+			int len;
+
+			len = Com_sprintf(libPath, sizeof(libPath), "%s%c%s", QUAKELIBDIR, PATH_SEP, name);
+			if(len < sizeof(libPath))
+			{
+				Com_Printf("Trying to load \"%s\" from \"%s\"...\n", name, QUAKELIBDIR);
+				dllhandle = Sys_LoadLibrary(libPath);
+			}
+			else
+			{
+				Com_Printf("Skipping trying to load \"%s\" from \"%s\", file name is too long.\n", name, QUAKELIBDIR);
+			}
+		}
+#endif
+
 		if(!dllhandle)
 		{
 			const char *basePath = Cvar_VariableString("fs_basepath");
