--- quake3/code/sys/sys_main.c~orig	2008-10-24 17:27:20.000000000 +0200
+++ quake3/code/sys/sys_main.c	2008-10-24 17:28:11.000000000 +0200
@@ -383,7 +383,8 @@
 Used to load a development dll instead of a virtual machine
 #1 look down current path
 #2 look in fs_homepath
-#3 look in fs_basepath
+#3 look in QUAKELIBDIR
+#4 look in fs_basepath
 =================
 */
 void *Sys_LoadDll( const char *name, char *fqpath ,
@@ -413,6 +414,11 @@
 	if(!libHandle && homepath)
 		libHandle = Sys_TryLibraryLoad(homepath, gamedir, fname, fqpath);
 
+#if defined(QUAKELIBDIR)
+	if(!libHandle)
+		libHandle = Sys_TryLibraryLoad(QUAKELIBDIR, gamedir, fname, fqpath);
+#endif
+
 	if(!libHandle && basepath)
 		libHandle = Sys_TryLibraryLoad(basepath, gamedir, fname, fqpath);
 
