--- quake3/Makefile	2005-12-05 02:38:48.000000000 +0100
+++ quake3.Makefile/Makefile	2005-12-05 17:42:51.000000000 +0100
@@ -151,9 +151,6 @@
 
   CC=gcc
 
-  ifeq ($(ARCH),alpha)
-    ARCH=axp
-  else
   ifeq ($(ARCH),x86_64)
     LIB=lib64
   else
@@ -165,7 +162,6 @@
   endif
   endif
   endif
-  endif
 
   BASE_CFLAGS = -Wall -fno-strict-aliasing -Wimplicit -Wstrict-prototypes -pipe
 
@@ -1654,6 +1650,26 @@
 	$(INSTALL) -s -m 0755 $(BR)/missionpack/ui$(ARCH).$(SHLIBEXT) \
 					$(COPYDIR)/missionpack/.
 
+install:
+	-$(MKDIR) -p -m 0755 $(BINDIR)/baseq3
+	$(INSTALL) -m 0755 $(BR)/ioquake3.$(ARCH)$(BINEXT) $(BINDIR)/quake3
+	$(INSTALL) -m 0755 $(BR)/ioquake3-smp.$(ARCH)$(BINEXT) $(BINDIR)/quake3-smp
+	$(INSTALL) -m 0755 $(BR)/ioq3ded.$(ARCH)$(BINEXT) $(BINDIR)/q3ded
+	-$(MKDIR) -p -m 0755 $(Q3LIBDIR)/baseq3
+	$(INSTALL) -m 0755 $(BR)/baseq3/cgame$(ARCH).$(SHLIBEXT) \
+					$(Q3LIBDIR)/baseq3/.
+	$(INSTALL) -m 0755 $(BR)/baseq3/qagame$(ARCH).$(SHLIBEXT) \
+					$(Q3LIBDIR)/baseq3/.
+	$(INSTALL) -m 0755 $(BR)/baseq3/ui$(ARCH).$(SHLIBEXT) \
+					$(Q3LIBDIR)/baseq3/.
+	-$(MKDIR) -p -m 0755 $(Q3LIBDIR)/missionpack
+	$(INSTALL) -m 0755 $(BR)/missionpack/cgame$(ARCH).$(SHLIBEXT) \
+					$(Q3LIBDIR)/missionpack/.
+	$(INSTALL) -m 0755 $(BR)/missionpack/qagame$(ARCH).$(SHLIBEXT) \
+					$(Q3LIBDIR)/missionpack/.
+	$(INSTALL) -m 0755 $(BR)/missionpack/ui$(ARCH).$(SHLIBEXT) \
+					$(Q3LIBDIR)/missionpack/.
+
 clean:clean-debug clean-release
 	$(MAKE) -C $(LOKISETUPDIR) clean
 
