diff -ur quake3/code/unix/Makefile quake3.alpha-fix/code/unix/Makefile
--- quake3/code/unix/Makefile	2005-10-15 18:02:11.000000000 +0200
+++ quake3.alpha-fix/code/unix/Makefile	2005-10-15 18:03:31.000000000 +0200
@@ -98,7 +98,7 @@
   CXX=g++
 
   ifeq ($(ARCH),alpha)
-    ARCH=axp
+    LIB=lib64
   else
   ifeq ($(ARCH),x86_64)
     LIB=lib64
diff -ur quake3/code/unix/unix_main.c quake3.alpha-fix/code/unix/unix_main.c
--- quake3/code/unix/unix_main.c	2005-10-15 15:34:29.000000000 +0200
+++ quake3.alpha-fix/code/unix/unix_main.c	2005-10-15 18:04:45.000000000 +0200
@@ -802,7 +802,7 @@
 #elif defined __sh__
   snprintf (fname, sizeof(fname), "%ssh.so", name);
 #else
-#error Unknown arch
+  snprintf (fname, sizeof(fname), "%sother.so", name);
 #endif
 
   // TODO: use fs_searchpaths from files.c
