--- quake3/code/unix/Makefile.orig	2005-10-31 02:33:28.121431424 +0100
+++ quake3/code/unix/Makefile	2005-10-31 02:49:44.691970136 +0100
@@ -91,9 +91,6 @@
   CC=gcc
   CXX=g++
 
-  ifeq ($(ARCH),alpha)
-    ARCH=axp
-  else
   ifeq ($(ARCH),x86_64)
     LIB=lib64
   else
@@ -105,7 +102,6 @@
   endif
   endif
   endif
-  endif
 
   BASE_CFLAGS =  -pipe -Wall -Wimplicit -Wstrict-prototypes
 
--- quake3/code/qcommon/q_platform.h.orig	2005-11-02 00:23:49.000000000 +0000
+++ quake3/code/qcommon/q_platform.h	2005-11-04 19:30:17.000000000 +0000
@@ -187,7 +187,7 @@
 
 #ifdef __i386__
 #define	CPUSTRING	"linux-i386"
-#elif defined __axp__
+#elif defined(__alpha__)
 #define	CPUSTRING	"linux-alpha"
 #elif defined __x86_64__
 #define	CPUSTRING	"linux-x86_64"
@@ -199,6 +199,8 @@
 #define	CPUSTRING	"linux-s390"
 #elif defined __s390x__
 #define	CPUSTRING	"linux-s390x"
+#elif defined(__sparc__)
+#define	CPUSTRING	"linux-sparc"
 #elif defined __ia64__
 #define	CPUSTRING	"linux-ia64"
 #else
